proto/common.proto

```diff
@@ -25,15 +25,23 @@ message SpanId {
 }
 
 message Field {
-    oneof field {
-        string debug = 1;
+    oneof name {
+        string str_name = 1;
+        uint64 name_idx = 2;
+    }
+    oneof value {
+        string debug_val = 3;
+        string str_val = 4;
+        uint64 u64_val = 5;
+        sint64 i64_val = 6;
+        bool bool_val = 7;
```

Comment:
nit: it would be nice to have some comments here explaining the different potential fields here. in particular, it would be nice to explain how names work, since that's potentially not clear?